### ISSUE TITLE
How to set the subviews height to superviews width?

### ISSUE THREAD

**Issue by abellee** [IjzerenHein/autolayout.js#7]
I think if I can use this like SnapKit?
and another question is how to set the subview's height equal superview's width
and subview's width equal superview's height?

**Comment by IjzerenHein** [comment]
Can you be more specific about your question regarding snapkit? I don't know snapkit myself, not sure what your question is about that.

Setting the subview's width to the superview's height is not supported directly through the syntax, but you can use a workaround for it. The thing is that you can't reference the superview directly, you can only use it at the beginning and end of a VFL line.

To workaround this, you can use a dummy subview (subview's prefixed with a _ can be considered dummy's) and make that equal to the width/height of the superview, like this:

```vfl
HV:|[_superviewDummy]|
H:|[myView(_superviewDummy.height)
V:|[myView(_superviewDummy.width)
```

**Comment by abellee** [comment]
I love Autolayout just like VFL of Apple so much, so I hope your project will be better and better.
SnapKit is a VFL lib of Swift, I can do autolayout just like 
```swift
contentView.snp_makeConstraints{(make)->Void in
       make.edges.equalTo(self.mainScrollView).offset(UIedgeInsetsMake(0, 0, -40, 0));
       make.width.equalTo(self.view);
}
```
in swift, and i think this will be more readable and easier for me on layout.

And I have used this lib in a small project, and I found some weird questions, just like if I used this autolayout for divs, then I wrote a pure div+css, seems like I can't put it in front of these divs which in autolayout, I think it maybe related to the z-index stack, but I tried so many ways, always can't change the stream of the layout with z-index, finally i have to autolayout it too, really makes me crazy.
